react/components/Dots.js

```diff
@@ -0,0 +1,147 @@
+import styles from './styles'
+import debounce from 'debounce'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+import React, { Component } from 'react'
+import { resolveSlidesNumber } from '../utils'
+import EventListener from 'react-event-listener'
+
+class Dots extends Component {
+  static propTypes = {
+    /** Classes to style the elements fo the component */
+    classes: PropTypes.shape({
+      root: PropTypes.string,
+      dot: PropTypes.string,
+      activeDot: PropTypes.string,
+      notActiveDot: PropTypes.string
+    }),
+    /** Extra props to be applied to the dot element */
+    dotProps: PropTypes.object,
+    /** The size of the dots, can be a number (in this case it will use px unit), or a string (you have to pass the number with the unit e.g '3rem') */
+    dotSize: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    /** Tag to be rendered in the dot element */
+    dotTag: PropTypes.string,
+    /** Function to change the currentSlide */
+    onChangeSlide: PropTypes.func.isRequired,
+    /** This prop works the same way the perPage of Slider and this component should receive the same value of Slider */
+    perPage: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
+    /** Tag of root element of the component to be rendered */
+    rootTag: PropTypes.string,
+    /** Total value of sliders that will be rendered */
+    totalSlides: PropTypes.number.isRequired,
+    /** If this flag is true, then every dot represent a page of slides (e.g. if perPage = 2 and you have 4 elements,
+     * then you have 2 dots), if false, then it will render a dot to every slide */
+    showDotsPerPage: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    classes: {
+      root: '',
+      dot: '',
+      activeDot: '',
+      notActiveDot: ''
+    },
+    dotTag: 'li',
+    perPage: 1,
+    rootTag: 'ul',
+    showDotsPerPage: false
+  }
+
+  constructor(props) {
```

Comment:
I think you don't need to use a constructor.

Reply:
I'm setting the `handleResize` down here